src/main/java/com/mongodb/jdbc/MongoSQLFunctions.java

```diff
@@ -0,0 +1,111 @@
+package com.mongodb.jdbc;
+
+public class MongoSQLFunctions extends MongoFunctions {
+    private static MongoSQLFunctions instance;
+
+    private MongoSQLFunctions(MongoFunction[] functions) {
+        super(functions);
+    }
+
+    public static MongoSQLFunctions getInstance() {
+        if (null == instance) {
+            instance =
+                    new MongoSQLFunctions(
+                            new MongoFunction[] {
+                                new MongoFunction(
+                                        "BIT_LENGTH",
+                                        "long",
```

Comment:
I'd like to see these repeated strings ie `BIT_LENGTH` and `long` as `protected static final String` in the abstract class like how we have it in `MongoDatabaseMetaData.java` and used for `MongoSQLFunctions.java` and `MySQLFunctions.java`.

Reply:
So I intentionally left the functions list completely separate because it seemed that they were quite different. BIT_LENGTH is not a supported MySQL function for example, so it does not make much sense to have it in the abstract class. I can refactor the common ones in the abstract class if you think that it will help readability.

Reply:
So I intentionally left the functions list completely separate because it seemed that they were quite different. BIT_LENGTH is not a supported MySQL function for example, so it does not make much sense to have it in the abstract class. I can refactor the common ones in the abstract class if you think that it will help readability.

For the type names, I don't see the point of having the same ones redefined in another file. They should be shared everywhere in the driver. After you merge SQL-559, then we can make use of it here for example and likely in the result set implementation too.